NewDawn/NewDawn/ViewControllers/ChatPageViewControllers/ChatRoomViewController.swift

```diff
@@ -352,16 +356,16 @@ class ChatRoomViewController: MessagesViewController {
                     alertController.setValue(messageText, forKey: "attributedMessage")
                     self.present(alertController, animated: true, completion: nil)
                     let confirmAction = UIAlertAction(title: "确定", style: .default) {(_) in
-                        HttpUtil.sendAction(user_from: self.userIdMe, user_to: self.userIdYou, action_type: UserActionType.REQUEST_TAKEN.rawValue, entity_type: EntityType.NONE.rawValue, entity_id: 0, message: UNKNOWN)
-                        let sentAlertController = UIAlertController(title: nil, message: "专属邀请已成功寄出。", preferredStyle: .alert)
-                        self.present(sentAlertController, animated: true, completion: nil)
-                        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
-                            sentAlertController.dismiss(animated: true, completion: nil)
+                        HttpUtil.sendAction(user_from: self.userIdMe, user_to: self.userIdYou, action_type: UserActionType.REQUEST_TAKEN.rawValue, entity_type: EntityType.NONE.rawValue, entity_id: 0, message: UNKNOWN) {
+                            success in
+                            if success {
+                                let sentAlertController = UIAlertController(title: nil, message: "专属邀请已成功寄出。", preferredStyle: .alert)
+                                self.present(sentAlertController, animated: true, completion: nil)
```

Comment:
I think you also need to wrap these lines into main queue. In general, you should wrap any UI code with main queue within callback function